diff --git a/plutus-core/src/Data/Aeson/THReader.hs b/plutus-core/src/Data/Aeson/THReader.hs
index 4b812b3..5290f87 100644
--- a/plutus-core/src/Data/Aeson/THReader.hs
+++ b/plutus-core/src/Data/Aeson/THReader.hs
@@ -5,10 +5,11 @@ import Data.Aeson
 import Language.Haskell.TH.Syntax
 import Language.Haskell.TH.Syntax.Compat
 import TH.RelativePaths
+import qualified Data.ByteString.Lazy as LBS

 readJSONFromFile :: (FromJSON a, Lift a) => String -> SpliceQ a
 readJSONFromFile name = liftSplice $ do
-    contents <- qReadFileLBS name
+    contents <- qRunIO $ LBS.readFile name
     case (eitherDecode contents) of
         Left err  -> fail err
         Right res -> examineSplice [||res||]
