From cbf132831293501a6a0402b5274e7aa2afff7509 Mon Sep 17 00:00:00 2001
From: Ken Micklas <git@kmicklas.com>
Date: Wed, 15 Apr 2020 20:43:49 -0400
Subject: [PATCH] Fix build on 32 bit platforms

---
 src/Data/TypeRepMap/Internal.hs | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/Data/TypeRepMap/Internal.hs b/src/Data/TypeRepMap/Internal.hs
index 880a230..ca09aa7 100644
--- a/src/Data/TypeRepMap/Internal.hs
+++ b/src/Data/TypeRepMap/Internal.hs
@@ -25,6 +25,8 @@ If you need to use this module for purposes other than tests,
 create an issue.
 -}
 
+#include "MachDeps.h"
+
 module Data.TypeRepMap.Internal where
 
 import Prelude hiding (lookup)
@@ -44,7 +46,13 @@ import Data.Semigroup (Semigroup (..), All(..))
 import GHC.Base (Any, Int (..), Int#, (*#), (+#), (<#))
 import GHC.Exts (IsList (..), inline, sortWith)
 import GHC.Fingerprint (Fingerprint (..))
+#if WORD_SIZE_IN_BITS >= 64
 import GHC.Prim (eqWord#, ltWord#)
+#else
+import GHC.IntWord64 (eqWord64#, ltWord64#)
+#define eqWord eqWord64
+#define ltWord ltWord64
+#endif
 import GHC.Word (Word64 (..))
 import Type.Reflection (SomeTypeRep (..), TypeRep, Typeable, typeRep, withTypeable)
 import Type.Reflection.Unsafe (typeRepFingerprint)
