From 85f36c88c16f1022e7b744a5422cc1ea62124342 Mon Sep 17 00:00:00 2001
From: Moritz Angermann <moritz.angermann@gmail.com>
Date: Fri, 8 May 2020 16:13:29 +0800
Subject: [PATCH] Update OFD.hsc

---
 src-ofd/Lukko/OFD.hsc | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src-ofd/Lukko/OFD.hsc b/src-ofd/Lukko/OFD.hsc
index 8d04dae..c835db8 100644
--- a/src-ofd/Lukko/OFD.hsc
+++ b/src-ofd/Lukko/OFD.hsc
@@ -109,8 +109,9 @@ hUnlock h = do
 -- Compat
 -------------------------------------------------------------------------------
 
--- there is no alignment in old hsc2hs
-#let alignmentcompat t = "%lu", (unsigned long)offsetof(struct {char x__; t (y__); }, y__)
+#if __GLASGOW_HASKELL__ < 800
+#  let alignment t = "%lu", (unsigned long)offsetof(struct {char x__; t (y__); }, y__)
+#endif
 
 -------------------------------------------------------------------------------
 -- implementation
@@ -128,7 +129,7 @@ data FLock  = FLock { l_type   :: CShort
 
 instance Storable FLock where
     sizeOf _ = #{size struct flock}
-    alignment _ = #{alignmentcompat struct flock}
+    alignment _ = #{alignment struct flock}
     poke ptr x = do
         fillBytes ptr 0 (sizeOf x)
         #{poke struct flock, l_type}   ptr (l_type x)
